oozie-to-airflow/utils/el_utils.py

```diff
@@ -162,14 +162,13 @@ def parse_els(properties_file: str, prop_dict: Dict[str, str] = None):
     return prop_dict
 
 
-def _convert_line(line: str, prop_dict: Dict[str, str]) -> None:
+def _convert_line(line: str, prop_dict: Dict[str, Union[str, list]]) -> None:
```

Comment:
I don't particularly like using Union of this kind. 

You never know what you get when you do prop_dict[key][0] (first element, or first character). I think it could have been better to always have list[str] as value and in case you need a single value you use prop_dict[key][0]. 

But I also understand that vast majority of cases will be string rather than list, so that's much better to use it this way, though we introduce an implicit coupling and typing does not really work well here.

Another option would be to continue keeping the array value as coma - separated string, which might be actually pretty clean and good solution. If there are just a few properties which are of type array, I find it pretty  ok to do the split(",") after retrieving the property from the dict.

Yet another option - we could have a utility function "get_as_array()" method used in places where you need to retrieve an array. I think that might be by far cleanest method.

We could also monkeypatch dict or add own subclass, but those are not really recommended ways in python.

It's just something to consider, I am ok with either solution eventually.

Reply:
Good call. I've changed it to keep it as a comma-separated string right up till the very end - it's now converted to a list right before being written out to a file.
See: https://github.com/GoogleCloudPlatform/cloud-composer/pull/45/commits/745f1066d125bb219191cc59c721e48581ff870d